ruby/ext/google/protobuf_c/message.c

```diff
@@ -153,9 +154,32 @@ static int extract_method_call(VALUE method_name, MessageHeader* self,
     accessor_type = METHOD_GETTER;
   }
 
+  bool has_field = upb_msgdef_lookupname(self->descriptor->msgdef, name, name_len,
+			                                   &test_f, &test_o);
+
+  // Look for enum accessor of the form <enum_name>_const
+  if (!has_field && accessor_type == METHOD_GETTER && name_len > 6) {
```

Comment:
Is this checking too loose?
What if another field's name is ```foo_XXXXX```?

Reply:
Please also test this case.

Reply:
I think it's okay because `METHOD_UNKNOWN` will still be returned unless the `enum_name` is found in the innermost check. I added that as a test case - is that what you were concerned about?

Reply:
I think it matters.
Suppose foo is the enum field, if you call m.foo_abcde, you don't return 
METHOD_UNKNOWN.

Reply:
Oh, I see. Good catch. I've added an additional check for the `_const` suffix.